xarray/core/dataset.py

```diff
@@ -1733,7 +1733,7 @@ def chunk(
         """
         from dask.base import tokenize
 
-        if isinstance(chunks, Number):
+        if is_scalar(chunks) and chunks is not None:
```

Comment:
Should this be
```suggestion
        if isinstance(chunks, (Number, str)):
```

or

```suggestion
        if isinstance(chunks, Number) or chunks == "auto":
```
?

Reply:
Thanks! I initially had used `if isinstance(chunks, Number) or chunks == "auto":` but then changed it after suggestions in #4055 , however I think it makes sense to revert now.

Reply:
I'll try using `None, Number, str, Mapping[Hashable, Union[None, Number, str, Tuple[Number, ...]]]` tomorrow :) thanks

Reply:
The mypy error probably stems from the fact that
```python
xr.core.utils.is_scalar(Mapping[Hashable, Union[None, Number, Tuple[Number, ...]]])
```
returns `True` (although I am not sure why).

Maybe you'll need `Literal["auto"]` instead of `str`.

Reply:
It seem that `isinstance(chunks, (Number, str))` is the more robust solution, because otherwise if only checking `chunks="auto"` then generic strings can still get through. 

I've also added `None, Number, str, Mapping[Hashable, Union[None, Number, str, Tuple[Number, ...]]]` and this appears to have fixed the `mypy` argument.

Edit: Sorry for all the extra commits, I keep forgetting to run the code format checks!

Reply:
don't worry about the commit history: only a single commit will end up in `master` since we squash all commits when merging a PR (as an example, you can take a look at #4012 which became 8834afa9f617bd201eba00374bb55d96dccec96b)